cpp/src/Slice/Grammar.y

```diff
@@ -222,19 +222,19 @@ definitions
 // ----------------------------------------------------------------------
 : definitions file_metadata
 {
-    auto metadata = dynamic_pointer_cast<StringListTok>($2);
+    auto metadata = dynamic_pointer_cast<MetadataListTok>($2);
     if (!metadata->v.empty())
     {
-        currentUnit->addFileMetadata(metadata->v);
+        currentUnit->addFileMetadata(std::move(metadata->v));
     }
 }
 | definitions metadata definition
 {
-    auto metadata = dynamic_pointer_cast<StringListTok>($2);
+    auto metadata = dynamic_pointer_cast<MetadataListTok>($2);
     auto contained = dynamic_pointer_cast<Contained>($3);
     if (contained && !metadata->v.empty())
     {
-        contained->setMetadata(metadata->v);
+        contained->setMetadata(std::move(metadata->v));
```

Comment:
Is this `std::move` correct, moving a data-meber out of the object seems uncommon. Is the object not used anymore after this?

Reply:
Yes, we never use the object again after this.

The only reason why we're using an object is because all the special `$N` types used by the parser must have a common base type (`GrammarBase`). So this object is just a type wrapper around the value that we're moving.